sanic/handlers.py

```diff
@@ -37,7 +40,18 @@ def _render_traceback_html(self, exception, request):
             uri=request.url)
 
     def add(self, exception, handler):
-        self.handlers[exception] = handler
+        self.handlers.append((exception, handler))
```

Comment:
Because there is no reason to keep it as a list. It is used only in iteration context.(at least in the error handler + other tests).
handlers as a list will prevent mistakes to accidentally access handlers by keys.